### DISCUSSION TITLE
How do I know which keys have been updated?

### DISCUSSION THREAD

**Discussion by lmk123** [pmndrs/valtio#1010] [Q&A] [answered]
I wanted to save the states, and then I saw [How to persist states](https://valtio.dev/docs/how-tos/how-to-persist-states) in the documentation, but I have a special case: my states can be huge, and I only want to save the really updated part.

For example:

```js
const state = proxy(
  readStorage() || {
    count: 0,
    text: 'hello',
  },
)

subscribe(state, (changes) => {
  // I hope changes will be `{ count: 123 }`, not the whole state `{ count: 123, text: 'hello' }`
  setStorage(changes)
})

state.count = 123
```

**Comment by dai-shi** [comment, marked as answer]
Valtio core doesn't track props, so you have to do it on your end.
`subscribeKey` is provided from `valtio/utils`, but it works only for a single key.

`subscribeKeys` is also proposed #887, but nobody has worked on since then.

---

We have `unstable_ops`, but it's considered for library authors or advanced users.

https://github.com/pmndrs/valtio/blob/0870cd109ad0e4a802be718272b7758f096821c0/src/vanilla.ts#L276
